[zstd] build failure

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Debian Sid WSL
- Compiler: em++

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install zstd --triplet=wasm32-emscripten`

-- [COMMUNITY] Loading triplet configuration from: /home/saltyfish/tools/vcpkg/triplets/community/wasm32-emscripten.cmake
-- Using cached facebook-zstd-v1.5.4.tar.gz.
-- Cleaning sources at /home/saltyfish/tools/vcpkg/buildtrees/zstd/src/v1.5.4-75b26c3dfe.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/saltyfish/tools/vcpkg/downloads/facebook-zstd-v1.5.4.tar.gz
-- Applying patch no-static-suffix.patch
-- Using source at /home/saltyfish/tools/vcpkg/buildtrees/zstd/src/v1.5.4-75b26c3dfe.clean
-- Found external ninja('1.11.1').
-- Configuring wasm32-emscripten
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/ninja -v
    Working Directory: /home/saltyfish/tools/vcpkg/buildtrees/zstd/wasm32-emscripten-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      /home/saltyfish/tools/vcpkg/buildtrees/zstd/config-wasm32-emscripten-dbg-CMakeCache.txt.log
      /home/saltyfish/tools/vcpkg/buildtrees/zstd/config-wasm32-emscripten-rel-CMakeCache.txt.log
      /home/saltyfish/tools/vcpkg/buildtrees/zstd/config-wasm32-emscripten-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:248 (vcpkg_execute_required_process)
  ports/zstd/portfile.cmake:15 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)

Additional context:
[config-wasm32-emscripten-out.log](https://github.com/microsoft/vcpkg/files/10908722/config-wasm32-emscripten-out.log)
[config-wasm32-emscripten-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10908732/config-wasm32-emscripten-dbg-CMakeCache.txt.log)
[config-wasm32-emscripten-rel-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10908729/config-wasm32-emscripten-rel-CMakeCache.txt.log)

--- a/ports/zstd/emscripten.patch
+++ b/ports/zstd/emscripten.patch
@@ -1,0 +1,14 @@
+diff --git a/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake b/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake
+index 0265349..1807d6e 100644
+--- a/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake
++++ b/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake
+@@ -54,7 +54,9 @@ macro(ADD_ZSTD_COMPILATION_FLAGS)
+         endif ()
+         # Add noexecstack flags
+         # LDFLAGS
++        if(NOT EMSCRIPTEN)
+         EnableCompilerFlag("-z noexecstack" false false true)
++        endif()
+         # CFLAGS & CXXFLAGS
+         EnableCompilerFlag("-Qunused-arguments" true true false)
+         EnableCompilerFlag("-Wa,--noexecstack" true true false)

--- a/ports/zstd/portfile.cmake
+++ b/ports/zstd/portfile.cmake
@@ -1,4 +1,3 @@
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebook/zstd
@@ -7,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF dev
     PATCHES
         no-static-suffix.patch
+        emscripten.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" ZSTD_BUILD_STATIC)

--- a/ports/zstd/vcpkg.json
+++ b/ports/zstd/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "zstd",
   "version": "1.5.4",
+  "port-version": 1,
   "description": "Zstandard - Fast real-time compression algorithm",
   "homepage": "https://facebook.github.io/zstd/",
   "license": "BSD-3-Clause OR GPL-2.0-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8674,7 +8674,7 @@
     },
     "zstd": {
       "baseline": "1.5.4",
-      "port-version": 0
+      "port-version": 1
     },
     "zstr": {
       "baseline": "1.0.7",

--- a/versions/z-/zstd.json
+++ b/versions/z-/zstd.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "4b61cbe1e314aa678d7cbf37a24b59e37694f4a6",
+      "version": "1.5.4",
+      "port-version": 1
+    },
+    {
       "git-tree": "16ff2b227a85c9f483480a6bb6d5eb0103ceee44",
       "version": "1.5.4",
       "port-version": 0

